### ISSUE TITLE
Chitu M21 serial connection

### ISSUE THREAD

**Issue by jpg0** [rudetrooper/Octoprint-Chituboard#42]
I have an Elegoo Mars 3 Pro which I have recently attempted to connected to octoprint. It is a newer ctbv4 firmware printer, so I am not expecting encrypted file support. I did however expect to be able to get serial communication working, but have failed to do so. The board is a Chitu M21, which doesn't seem to be used elsewhere. It does have plainly marked pins, however I have not been able to get any serial communication going, after connecting the GND, TX, and RX pins as in the photo.

Any suggestions? 

Thanks

![IMG_20230320_210337](https://user-images.githubusercontent.com/709622/226317702-f8c1e08e-37fd-4b9b-8a09-c3f07df8ddb0.jpg)

**Comment by jpg0** [comment]
I also tried on both a pi and a macbook, also with varying the baud rate. I'm fairly sure the voltage is correct as the rx and tx pins are both at 3.3V, but there is no comms happening. I suspect that Chitu have disabled the serial communication.

**Comment by bvaerewyck** [comment]
Did you reverse the tx and rx?  I mad that mistake a few years ago.  Sorry if I missed that and you already tried.  Hope it works.

**Comment by jpg0** [comment]
Yeah, I tried both ways, lots of times. Pretty sure there is nothing coming out.

**Comment by bvaerewyck** [comment]
Is it the adafruit cable?  Did you test continuity?  Bad cable?

**Comment by jpg0** [comment]
I used some random cable that I had, although I did test continuity with a multimeter beforehand and also tried more than one cable (including a different cable attached to a usb ftdi adapter connected to a laptop).

One of the reasons that I posted here was to see if others saw the same behaviour as me to confirm whether I had made a mistake or there is no comms. So far I am aware of 2 (myself included) people who see no signal, and 0 who see a signal!

I suspect though that we will see more people try this though, as the Mars 3 Pro is fairly new and seems like a pretty good piece of hardware.

**Comment by linux-paul** [comment]
I've also teted differtent cables with no effect, Also I tested boudrates up to 1000000.
but ... depending to the fact of the 5VVcc pin and the chip has been changed from ARM controller to FPGA ... maybe a TTL converter will be helpfull in thi case?

**Comment by jpg0** [comment]
Well I did check the voltage on the TX and RX pins with a multimeter and they were 3.3V. I can certainly try listening at 5V, although wouldn't you have expected to see _something_ also if listening at 3.3V?

**Comment by bvaerewyck** [comment]
I have 9 printers and I always use this cable...  https://www.amazon.com/ADAFRUIT-Industries-954-Serial-Raspberry/dp/B00DJUHGHI/ref=sr_1_1?crid=JCJOJDDYODR7&keywords=adafruit+ttl+cable&qid=1680056620&s=industrial&sprefix=adafruit+ttl+cable%2Cindustrial%2C148&sr=1-1  works without any issues.

**Comment by linux-paul** [comment]
@bvaerewyck If you are running a chituboard serial over usb, which pi board are in use? My A3+ just has one usb port, which is configured as dwc usb drive gadget after installation. But driving the printerboard serial with TTL signals meets my expectation and maybe I'll receive my TTL bridges tomorrow.

**Comment by linux-paul** [comment]
levelshifter does also not help and .... btw .... the ARM chip has been placed on the backside ...

**Comment by jpg0** [comment]
@linux-paul there were no other connection points on the backside were there? I never pulled out my board to check the other side.

**Comment by linux-paul** [comment]
I didn't pulled it out completly, just so far to get a quick view for looking to adidional connectors. There are a pretty bunch of chips solderd on the backside, It looks like completely soldered except one chip close to the pinheader. There are no further connectors, so far I think.

**Comment by zackfuchtel** [comment]
> I also tried on both a pi and a macbook, also with varying the baud rate. I'm fairly sure the voltage is correct as the rx and tx pins are both at 3.3V, but there is no comms happening. I suspect that Chitu have disabled the serial communication.

I also think it was disabled. Not a single byte to receive.
